tools/buildgen/extract_metadata_from_bazel_xml.py

```diff
@@ -139,10 +154,17 @@ def _extract_nonpublic_headers(bazel_rule):
 def _extract_sources(bazel_rule):
```

Comment:
right now in the "tools/buildgen/extract_metadata_from_bazel_xml.py" script, the logic for reading the build metadata from the bazel build and converting it into something we're later going to process contains no "custom" (or "project specific") logic - it simply extracts the source files, headers etc. 
Only after we've extracted this general information, we go ahead a postprocess it in a way that's grpc-project specific. So I'd rather not include custom handling logic like "if RE_SUPPORTED_LIB.match(xyz)" in general functions like "_extract_sources" (there should be nothing custom about "Gets list of source files from a bazel rule").